src/source/RazorEngine.Core/Templating/IsolatedRazorEngineService.cs

```diff
@@ -87,7 +87,16 @@ public DefaultConfigCreator()
             /// <returns></returns>
             public ITemplateServiceConfiguration CreateConfiguration()
             {
-                return new TemplateServiceConfiguration();
+                // We need to use the DefaultCompilerServiceFactory because we
+                // get conflicting TypeLoadExceptions in RazorEngineSourceReferenceResolver
+                // that cannot be resolved.
+                // ie) When not used inside the IsolatedRazorEngineService, we
+                // need [SecuritySafeCritical] for the methods. However, inside
+                // the Isolated service, it requires everything to be [SecurityCritical].
```

Comment:
I've created a branch here where I use Roslyn as the default for you to check out the weird security critical issue I was having when running Debug|NET45.
[conniey/RazorEngine/roslynAsDefault](https://github.com/conniey/RazorEngine/tree/roslynAsDefault)

**Test:** IsolatedRazorEngineService_Sandbox_WithISerializableModel

```
Message: System.TypeLoadException : Inheritance security rules violated while overriding member: 
'RazorEngine.Roslyn.CSharp.RoslynCompilerServiceBase+RazorEngineSourceReferenceResolver.NormalizePath(System.String, System.String)'. 
Security accessibility of the overriding method must match the security accessibility of the method being overriden.
```

So, I change all the attributes in [RoslynCompilerServiceBase+RazorEngineSourceReferenceResolver](https://github.com/conniey/RazorEngine/blob/536686b3c2d217b2ee501a6e41cf74a2395d68a0/src/source/RazorEngine.Core/Compilation/RoslynCompilerServiceBase.cs#L103) from [SecurityCritical] to [SecuritySafeCritical].  Methods changed:
* `RazorEngineSourceReferenceResolver.NormalizePath(System.String, System.String)`
* `RazorEngineSourceReferenceResolver.OpenRead(System.String)`
* `RazorEngineSourceReferenceResolver.ResolveReference(System.String, System.String)`

And then I get another issue which may be caused by this [issue in Microsoft.CodeAnalysis 1.3.2](https://stackoverflow.com/a/39785966/4220757):

```
System.MissingMethodException : 
Method not found: 
'System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.Diagnostic> Microsoft.CodeAnalysis.Emit.EmitResult.get_Diagnostics()'.
```

and as a result of those changes, all the other tests that used to pass (ie. AssemblyIsScannable), fail with this exception:
```
Message: System.Reflection.ReflectionTypeLoadException : Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
  ----> System.TypeLoadException : Inheritance security rules violated while overriding member: 'RazorEngine.Roslyn.CSharp.RoslynCompilerServiceBase+RazorEngineSourceReferenceResolver.NormalizePath(System.String, System.String)'. Security accessibility of the overriding method must match the security accessibility of the method being overriden.
```

Any help or tips would be super awesome. :( I am probably not understanding the security model.